cdap-ui/app/cdap/components/CaskWizards/AddNamespace/index.js

```diff
@@ -36,8 +37,14 @@ export default class AddNamespaceWizard extends Component {
   createNamespace() {
     return PublishNamespace()
       .flatMap(
-        () => {
-          return PublishPreferences();
+        (res) => {
+          if (res.includes('already exists')) {
+            return Rx.Observable.create((observer) => {
```

Comment:
I tried doing that, but other functions that inherit this one as a prop (e.g. `submitForm` in the `Wizard` component) expect the return value to be an observable, so it didn't work.